core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -988,6 +998,10 @@ void cleanupExpiredRemoteLogSegments() throws RemoteStorageException, ExecutionE
                         return;
                     }
                     RemoteLogSegmentMetadata metadata = segmentsIterator.next();
+
+                    if (!SEGMENT_DELETION_VALID_STATES.contains(metadata.state())) {
```

Comment:
Can this condition be simplified to?

```
if (RemoteLogSegmentState.DELETE_SEGMENT_FINISHED.equals(metadata.state())) {
    continue;
}
```

If we want to add/remove state, we can change the logic later.

Reply:
Good idea, applying change.